### ISSUE TITLE
Kancolle-auto crash on check expedition time

### ISSUE THREAD

**Issue by Kiras756** [mrmin123/kancolle-auto#343]
#### Environment details
* kancolle-auto version: 2017-03-17 (Release 9)
* Viewer/browser: Opera
* Operating system: Windows XP SP3

#### Issue
Consistent (can be reproduced consistently).

Today I'm updated my old Kancolle-Auto to actual version(2017-03-17 (Release 9)) and also updated Sikuli to a new 1.1.1 version because you know why. And then, when bot checks expedition time - occurs a crash. Recently, many time ago, I faced with this problem and the cause of this was missed library. Now I don't know why Kancolle-auto doesn't work.
ps
Sikuli website missed with recommended Sikuli build... mm, maybe this a reason? Anyone can share recommended build?

All logs attached

[log.txt](https://github.com/mrmin123/kancolle-auto/files/853179/log.txt)
[hs_err_pid2808.txt](https://github.com/mrmin123/kancolle-auto/files/853181/hs_err_pid2808.txt)

pss
Sorry for my bad english =)

**Comment by stackhanovets** [comment]
@Kiras756,
Did you try to update your JRE to version 8? Also please check [this link](https://oss.sonatype.org/content/groups/public/com/sikulix/sikulixsetup/1.1.1-SNAPSHOT/sikulixsetup-1.1.1-20170318.163703-92-forsetup.jar).

**Comment by Kiras756** [comment]
I tried different Java version, include and last.
Thx for link, testing soon.

**Comment by mrmin123** [comment]
Considering where it's failing, it might be something to do with the tesseract library. Starting with 1.1.1 I know it gets installed somewhere else and the Sikuli installer is supposed to be more reliable in installing it, but that'd be the first thing I look at... Trying the updated version @stackhanovets linked might help. I'm personally on [sikulixsetup-1.1.1-20170115](https://oss.sonatype.org/content/groups/public/com/sikulix/sikulixsetup/1.1.1-SNAPSHOT/sikulixsetup-1.1.1-20170115.001619-83-forsetup.jar) which is slightly older.

**Comment by Kiras756** [comment]
@stackhanovets hm, this version I already have and it don't work
@mrmin123 link is broken

**Comment by waicool20** [comment]
The sikulix binary is failing with a call to the Tesseract libraries, by principal the library files are included in the sikulix.jar so I reckon just reinstalling sikulix and it should work fine. Provided that it reinstalls correctly.

Though one thing I don't get is that your log.txt reports a java version of ```7.0_13-b20``` on the other hand your hs_err_pid2808.txt reports ```8.0_121-b13``` Might be hinting a java problem (Nuke it and reinstall perhaps?)

**Comment by stackhanovets** [comment]
@Kiras756 , 
tl;dr update to win10

I've also recollected some issues on Windows XP. Particularly, a shouha/chuuha differentiation problem in the repair module. When two ships have close HP but one has 小破 icon and second is marked by 中破  the repair module just stucks and clicks within the docks. I don't know what's wrong with the image recognition, color depth or VMWare XP video driver, After testing it seems to be no problem on Debian or Windows 10.

**Comment by Kiras756** [comment]
@waicool20 
>Though one thing I don't get is that your log.txt reports a java version of 7.0_13-b20 on the other hand your hs_err_pid2808.txt reports 8.0_121-b13 Might be hinting a java problem (Nuke it and reinstall perhaps?)

sorry, log from 2 different test, but be sure, it all the same always

**Comment by Kiras756** [comment]
@stackhanovets 
>tl;dr update to win10

maybe in another life
btw, this is my "server" on old hardware

**Comment by Kiras756** [comment]
@waicool20 
>The sikulix binary is failing with a call to the Tesseract libraries, by principal the library files are included in the sikulix.jar so I reckon just reinstalling sikulix and it should work fine. Provided that it reinstalls correctly.

Reinstalled Sikuli 5 or 6 times - no effect

**Comment by stackhanovets** [comment]
@Kiras756 OMG you just gave me an idea to test Sikuli features on Raspberry Pi. Good luck then.

**Comment by waicool20** [comment]
[Just tried this one right now](https://oss.sonatype.org/content/groups/public/com/sikulix/sikulixsetup/1.1.1-SNAPSHOT/sikulixsetup-1.1.1-20170319.001620-93-forsetup.jar) and OCR is working fine for me. It's one version newer than the one above. [Otherwise have fun trying one of these till one works.](https://oss.sonatype.org/content/groups/public/com/sikulix/sikulixsetup/1.1.1-SNAPSHOT)

Another option is to open up the generated jar file and see if tesseract files are actually being stuffed inside.

This is what I got relative to the root inside the Jar file:
>$ find . | grep "tess"
./srcnativelibs/Include/Tesseract/tesseract
./srcnativelibs/Include/Tesseract/tesseract/tesscallback.h
./srcnativelibs/Include/Tesseract/tesseract/renderer.h
./srcnativelibs/Include/Tesseract/tesseract/errcode.h
./srcnativelibs/Include/Tesseract/tesseract/unicharset.h
./srcnativelibs/Include/Tesseract/tesseract/basedir.h
./srcnativelibs/Include/Tesseract/tesseract/osdetect.h
./srcnativelibs/Include/Tesseract/tesseract/capi.h
./srcnativelibs/Include/Tesseract/tesseract/unicharmap.h
./srcnativelibs/Include/Tesseract/tesseract/pageiterator.h
./srcnativelibs/Include/Tesseract/tesseract/platform.h
./srcnativelibs/Include/Tesseract/tesseract/serialis.h
./srcnativelibs/Include/Tesseract/tesseract/strngs.h
./srcnativelibs/Include/Tesseract/tesseract/thresholder.h
./srcnativelibs/Include/Tesseract/tesseract/memry.h
./srcnativelibs/Include/Tesseract/tesseract/unichar.h
./srcnativelibs/Include/Tesseract/tesseract/params.h
./srcnativelibs/Include/Tesseract/tesseract/helpers.h
./srcnativelibs/Include/Tesseract/tesseract/resultiterator.h
./srcnativelibs/Include/Tesseract/tesseract/fileerr.h
./srcnativelibs/Include/Tesseract/tesseract/apitypes.h
./srcnativelibs/Include/Tesseract/tesseract/genericvector.h
./srcnativelibs/Include/Tesseract/tesseract/ocrclass.h
./srcnativelibs/Include/Tesseract/tesseract/baseapi.h
./srcnativelibs/Include/Tesseract/tesseract/host.h
./srcnativelibs/Include/Tesseract/tesseract/publictypes.h
./srcnativelibs/Include/Tesseract/tesseract/ndminx.h
./srcnativelibs/Include/Tesseract/tesseract/ltrresultiterator.h
./srcnativelibs/Vision/tessocr.cpp
./srcnativelibs/Vision/tessocr.h
./sikulixtessdata
./sikulixtessdata/eng.cube.nn
./sikulixtessdata/eng.cube.lm
./sikulixtessdata/sikulixcontent
./sikulixtessdata/eng.traineddata
./sikulixtessdata/eng.cube.fold
./sikulixtessdata/eng.cube.params
./sikulixtessdata/eng.cube.bigrams
./sikulixtessdata/configs
./sikulixtessdata/configs/strokewidth
./sikulixtessdata/configs/logfile
./sikulixtessdata/configs/rebox
./sikulixtessdata/configs/linebox
./sikulixtessdata/configs/box.train
./sikulixtessdata/configs/unlv
./sikulixtessdata/configs/kannada
./sikulixtessdata/configs/digits
./sikulixtessdata/configs/makebox
./sikulixtessdata/configs/inter
./sikulixtessdata/configs/hocr
./sikulixtessdata/configs/api_config
./sikulixtessdata/configs/quiet
./sikulixtessdata/configs/box.train.stderr
./sikulixtessdata/configs/ambigs.train
./sikulixtessdata/eng.cube.word-freq
./sikulixtessdata/tessconfigs
./sikulixtessdata/tessconfigs/segdemo
./sikulixtessdata/tessconfigs/matdemo
./sikulixtessdata/tessconfigs/msdemo
./sikulixtessdata/tessconfigs/batch
./sikulixtessdata/tessconfigs/batch.nochop
./sikulixtessdata/tessconfigs/nobatch
./sikulixtessdata/eng.cube.size
./sikulixtessdata/eng.tesseract_cube.nn

Results may vary since obviously I'm on Linux, but you should check if the general file structure is something like above

**Comment by Kiras756** [comment]
@waicool20 I have this
![1](https://cloud.githubusercontent.com/assets/26521625/24082934/ef6c3efa-0cde-11e7-8d55-d822e06eb9bc.png)
![2](https://cloud.githubusercontent.com/assets/26521625/24082935/f3da8744-0cde-11e7-820c-702d79d674d3.png)
![2_1](https://cloud.githubusercontent.com/assets/26521625/24082936/f6bd23fe-0cde-11e7-8bb5-1a3751cb18e2.png)
![2_2](https://cloud.githubusercontent.com/assets/26521625/24082939/fa5e28be-0cde-11e7-8506-9ce23c6b769a.png)
![3](https://cloud.githubusercontent.com/assets/26521625/24082940/fca377f0-0cde-11e7-80ed-616335a1f3de.png)

**Comment by waicool20** [comment]
Well considering that the libtesseract-3.dll is in your archive, I don't see any reason for it to NOT work. Final ditch option is to manually install tesseract to your system so it doesn't need to use the internal jar dll. My jar file doesn't have the libtesseract-3 file because I put all my tesseract files in PATH.

@mrmin123 On the other hand, I find that the link to the sikulixsetup in the README.md is dead, so that has to be updated. Also while you're at it maybe change it to include some kind of basic troubleshooting by increasing ```SleepModifier``` to maybe up to limit of 10 for single-shot type problems to reduce the similar types of recent problems that could have been fixed by doing just that. The FAQ just mentions crashes so people might not even bother trying.

**Comment by waicool20** [comment]
Btw it's never to late to convert to a Linux user, especially for old server hardware ;)

**Comment by Kiras756** [comment]
>Well considering that the libtesseract-3.dll is in your archive, I don't see any reason for it to NOT work. Final ditch option is to manually install tesseract to your system so it doesn't need to use the internal jar dll. My jar file doesn't have the libtesseract-3 file because I put all my tesseract files in PATH.

How to do this?

**Comment by waicool20** [comment]
Not exactly sure since I never have tried a windows install, this came up on [google](https://github.com/TesseractTeam/Tesseract/wiki/Installation-on-Windows), might be worth a shot.

So far what I've seen, other than the guide above is that you'll have to compile tesseract from source which is VERY not fun especially on windows.

**Comment by Kiras756** [comment]
>Not exactly sure since I never have tried a windows install, this came up on google, might be worth a shot.

Shit. Link for step#3 from guide doesn't work. Today is totally not my day =(

**Comment by waicool20** [comment]
lol I don't think that matters, I just noticed that that tesseract(some Minecraft thing) is not the right tesseract(OCR), so I guess don't even bother. This should be the right one lol sorry derp:

https://github.com/tesseract-ocr/tesseract/wiki

Best of luck :)

**Comment by waicool20** [comment]
Your in luck actually haha since the above link seems to have an installer for windows :)

**Comment by Kiras756** [comment]
Doesn't help, still crash and exactly the same error =(

**Comment by Kiras756** [comment]
Ok, I don't need a new futures, if I just replace old images with a new ones in my old Kancolle-Auto it will be work fine?

**Comment by stackhanovets** [comment]
@Kiras756 I've just tried pre-SikuliX release with updated images and it has worked almost perfectly, excepting some described issues. However, there were _Sikuli, libs/tesseract_ etc. in my local copy folder included and untouched. If you did not care about the backups, there is no guarantees everything could be fine. Please look for Sikuli troubleshooting guide after all.

@Kiras756 I had a deal with `SleepModifier` parameter, and the resupplying bug on WinXP persisted upon present. I don't care so much about it because the module was working fine on Debian/Win10 with the very same config, so I've got alternatives already.

**Comment by mrmin123** [comment]
I can't help much with this since the issue has to do with tesseract/Sikuli/maybe even OS-specific.

Leaving this open as a reminder to implement changes to readme as suggested by @waicool20

**Comment by Kiras756** [comment]
Okay, I found a solution for my problem!
Tesseract does not eat сyrillic alphabet on path to self. Sikuli 1.1.1+ extract some libs into %appdata% and execute from there. In my case, my profile have a сyrillic name. I'm tested Kancolle-Auto on another profile and all work perfectly! Then replaced on old profile %appdata% from %userprofile% (with some registry-edit) and... All work!
Issue completely resolved!
